### PR TITLE
#0: add device perf ci tests for SD

### PR DESCRIPTION
**Pull request by mtatsumiTT**
all post-commit pipeline passes([link](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8622201263)), and added tests pass on [device-perf](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8622196945/job/23632807803) and [model-perf](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8622199300/job/23632816663).

### CODE REVIEW

**Comment by tt-rkim** [comment]
will merge for you once this passes and shows the results we want: https://github.com/tenstorrent-metal/tt-metal/actions/runs/8570989816/job/23490220770

**Comment by mtatsumiTT** [comment]
sorry I just noticed that I pushed wrong commits ... it would be great if you can check the results of this [model-perf](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8571457689) and [device-perf](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8571461162)

**Comment by tt-rkim** [comment]
Were they wrong? Note taht we don't run device perf regressions on WH just yet, but you added the definition. You can re-add it back and I will review

**Comment by mtatsumiTT** [comment]
oh ok, I'll add it back and re-run device-perf then. The previous one I pushed was missing other changes anyway, I'll add the change to current one.

**Review by arakhmati — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by boris-drazic — APPROVED** [review]

File: tt_metal/core_descriptors/wormhole_b0_80_arch_eth_dispatch.yaml [hunk 1]
```diff
@@ -61,7 +61,7 @@ nebula_x1:
 nebula_x2:
   1:
     l1_bank_size:
-      749568
+      1376256
```
Comment (mtatsumiTT) [inline tt_metal/core_descriptors/wormhole_b0_80_arch_eth_dispatch.yaml:64]: fyi @abhullar-tt

**Review by tt-rkim — APPROVED** [review]

File: tests/scripts/run_performance.sh [hunk 2]
```diff
@@ -88,6 +92,8 @@ run_device_perf_models() {
```
Comment (mo-tenstorrent) [inline tests/scripts/run_performance.sh:92]: It looks like some the tests on this list are not managing device switch pretty well. 

For example bert is being attempted here: 
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8622196945/job/23632807803#step:7:1367
and it is crashing. Even if it wasn't crashing its expected numbers are for GS and not WHB0. 

I think we should check for device arch here as well, similar to what you have done on line 75 for models.

Comment (mo-tenstorrent) [reply tests/scripts/run_performance.sh:92]: Or we can tag all the test here properly. This might actually be the cleaner option.

Comment (mo-tenstorrent) [reply tests/scripts/run_performance.sh:92]: Sounds good yeah thanks

File: tests/scripts/run_performance.sh [hunk 3]
```diff
@@ -84,17 +88,21 @@ run_device_perf_models() {
     #Model Device perf regression tests to make sure thy run on no-soft-reset BMs
     tests/scripts/run_profiler_regressions.sh PROFILER
```
Comment (mo-tenstorrent) [inline tests/scripts/run_performance.sh:89]: This should be under GS as well. WH does this test as part of post commit already.

**Comment by mtatsumiTT** [comment]
related pipelines ([device-perf B0](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8652244795/job/23726711561), [model-perf cnn-javelin](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8652179065), and [post-commit](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8652184440)) pass, I'll merge it.

**Pull request merged**